web/src/user_profile.ts

```diff
@@ -607,45 +692,56 @@ export function show_edit_bot_info_modal(user_id, $container) {
         is_incoming_webhook_bot: bot.bot_type === INCOMING_WEBHOOK_BOT_TYPE,
     });
     $container.append($(html_body));
-    let avatar_widget;
+    let avatar_widget: UploadWidget;
+
+    assert(bot.bot_type !== undefined && bot.bot_type !== null);
 
     const bot_type = bot.bot_type.toString();
-    const service = bot_data.get_services(bot.user_id)[0];
+    const service = bot_data.get_services(bot.user_id)![0];
```

Comment:
Is `service` always present? I expect an incoming webhook bot to not have one.

Reply:
Can you make comments saying how you addressed the feedback, in cases like this where it's not clear what is correct?

Reply:
Yes, for the `incoming webhook bot` the service shouldn't be present and I've handled that case in the latest updates.